Student/Student/Submissions/SubmissionRubric/RubricCircleView.swift

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright (C) 2019-present Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import Core
+
+class RubricCircleView: UIView {
+    private static let w: CGFloat = 49
+    private static let space: CGFloat = 10
+    var rubric: RubricViewModel? {
+        didSet {
+            setNeedsDisplay()
+        }
+    }
+
+    private static var formatter: NumberFormatter = {
+        var formatter = NumberFormatter()
+        formatter.roundingIncrement = 0.01
+        formatter.numberStyle = .decimal
+        return formatter
+    }()
+
+    static func computedHeight(rubric: RubricViewModel, maxWidth: CGFloat) -> CGFloat {
+        let count = CGFloat(rubric.ratings.count)
+        let howManyCanFitInWidth = CGFloat( floor( maxWidth / (w + space) ) )
+        let rows = CGFloat(ceil(count / howManyCanFitInWidth))
+        return (rows * w) + ((rows - 1) * space)
+    }
+
+    override func draw(_ rect: CGRect) {
```

Comment:
Would a view per circle work better for this? The RN view it is related to has buttons for each circle (maybe only needed for teacher app unless peer reviews use them) and there is also a tooltip displayed with the rating description, right?

Reply:
I did this intentionally to help make the scrolling smooth and reduce the amount of subviews in each cell.  I'm unaware of any tooltips.  To my understanding in student, there is not much interaction with these. I'll double check with Dave and Peyton.